src/ray/gcs/redis_accessor.h

```diff
@@ -132,6 +132,48 @@ class RedisTaskInfoAccessor : public TaskInfoAccessor {
   TaskSubscriptionExecutor task_sub_executor_;
 };
 
+/// \class RedisObjectInfoAccessor
+/// RedisObjectInfoAccessor is an implementation of `ObjectInfoAccessor`
+/// that uses Redis as the backend storage.
+class RedisObjectInfoAccessor : public ObjectInfoAccessor {
+ public:
+  explicit RedisObjectInfoAccessor(RedisGcsClient *client_impl);
+
+  virtual ~RedisObjectInfoAccessor() {}
+
+  Status AsyncGetLocations(const ObjectID &object_id,
+                           const MultiItemCallback<ObjectTableData> &callback) override;
+
+  Status AsyncAddLocation(const ObjectID &object_id, const ClientID &node_id,
+                          const StatusCallback &callback) override;
+
+  Status AsyncRemoveLocation(const ObjectID &object_id, const ClientID &node_id,
+                             const StatusCallback &callback) override;
+
+  Status AsyncSubscribeToLocations(
+      const ObjectID &object_id,
+      const SubscribeCallback<ObjectID, ObjectChangeNotification> &subscribe,
+      const StatusCallback &done) override;
+
+  Status AsyncUnsubscribeToLocations(const ObjectID &object_id,
+                                     const StatusCallback &done) override;
+
+ private:
+  RedisGcsClient *client_impl_{nullptr};
+
+  // Use a random ClientID for object subscription. Because:
+  // If we use ClientID::Nil, GCS will still send all objects' updates to this GCS Client.
+  // Even we can filter out irrelevant updates, but there will be extra overhead.
+  // And because the new GCS Client will no longer hold the local ClientID, so we use
+  // random ClientID instead.
+  // TODO(micafan): Remove this random id, once GCS becomes a service.
+  ClientID node_id_{ClientID::FromRandom()};
```

Comment:
There are multiple random node ids in different accessors. Can you unify them? E.g., only having one in `RedisGcsClient`.
Also, it's named `subscribe_id_` elsewhere, which sounds more aligned with its purpose.

Reply:
Already rename it to `subscribe_id_`. 
If we unify all the accessor's `random id`, then we might change the `constructor` of those `accessor`(pass the random id), or add a new `interface for random id` to `RedisGcsClient`, either way it will change the interface.  I prefer to make `subscribe id` as the internal implementation.